alerting/src/main/kotlin/org/opensearch/alerting/workflow/CompositeWorkflowRunner.kt

```diff
@@ -80,87 +102,146 @@ object CompositeWorkflowRunner : WorkflowRunner() {
-                        delegateMonitor,
-                        monitorCtx,
-                        periodStart,
-                        periodEnd,
-                        dryRun,
-                        workflowRunContext
-                    )
-                } else if (delegateMonitor.isDocLevelMonitor()) {
-                    DocumentLevelMonitorRunner.runMonitor(
-                        delegateMonitor,
-                        monitorCtx,
-                        periodStart,
-                        periodEnd,
-                        dryRun,
-                        workflowRunContext
-                    )
-                } else if (delegateMonitor.isQueryLevelMonitor()) {
-                    QueryLevelMonitorRunner.runMonitor(
-                        delegateMonitor,
-                        monitorCtx,
-                        periodStart,
-                        periodEnd,
-                        dryRun,
-                        workflowRunContext
-                    )
-                } else {
-                    throw AlertingException.wrap(
-                        IllegalStateException("Unsupported monitor type")
-                    )
-                }
+                dataSources = delegateMonitor.dataSources
+                val delegateRunResult =
+                    runDelegateMonitor(delegateMonitor, monitorCtx, periodStart, periodEnd, dryRun, workflowRunContext, executionId)
+                resultList.add(delegateRunResult!!)
             } catch (ex: Exception) {
-                logger.error("Error executing workflow delegate. Error: ${ex.message}", ex)
+                logger.error("Error executing workflow delegate monitor ${delegate.monitorId}", ex)
                 lastErrorDelegateRun = AlertingException.wrap(ex)
-                continue
+                break
             }
-            if (delegateRunResult != null) resultList.add(delegateRunResult)
         }
-        logger.debug("Workflow ${workflow.id} in $executionId finished")
+        logger.debug("Workflow ${workflow.id} delegate monitors in execution $executionId completed")
         // Update metadata only if the workflow is not temp
         if (!isTempWorkflow) {
             WorkflowMetadataService.upsertWorkflowMetadata(
                 workflowMetadata.copy(latestRunTime = workflowExecutionStartTime, latestExecutionId = executionId),
                 true
             )
         }
+        val triggerResults = mutableMapOf<String, ChainedAlertTriggerRunResult>()
+        val workflowRunResult = WorkflowRunResult(
+            workflowId = workflow.id,
+            workflowName = workflow.name,
+            monitorRunResults = resultList,
+            executionStartTime = workflowExecutionStartTime,
+            executionEndTime = null,
+            executionId = executionId,
+            error = lastErrorDelegateRun,
+            triggerResults = triggerResults
+        )
+        if (dataSources != null) {
```

Comment:
Why are we doing this only when `dataSources != null`?

Reply:
dataSources is never null. will always default to AlertIndices, FindingIndex, QueryIndex defined in Alerting plugin

Reply:
Then why do we do this check?